### ISSUE TITLE
More meaningful error messages when you cannot connect

### ISSUE THREAD

**Issue by box293** [NagiosEnterprises/nrpe#166]
Here is when I try and connect to a remote host using `check_nrpe` for which I do not have permission:
```
$ check_nrpe -4 -H centos12
CHECK_NRPE: Error - Could not connect to 10.25.13.30: Connection reset by peer
```

If I look at the logs on the remote host I see:
```
Sep 13 11:10:06 centos12 nrpe[1677]: Starting up daemon
Sep 13 11:10:06 centos12 nrpe[1677]: Server listening on 0.0.0.0 port 5666.
Sep 13 11:10:06 centos12 nrpe[1677]: socket: Address family not supported by protocol
Sep 13 11:10:06 centos12 nrpe[1677]: Warning: Daemon is configured to accept command arguments from clients!
Sep 13 11:10:06 centos12 nrpe[1677]: Listening for connections on port 5666
Sep 13 11:10:06 centos12 nrpe[1677]: Allowing connections from: 127.0.0.1
Sep 13 11:10:08 centos12 nrpe[1683]: Host 10.25.254.50 is not allowed to talk to us!
```

Is there any way the `Host 10.25.254.50 is not allowed to talk to us!` message could be returned to check_nrpe?
```
CHECK_NRPE: Error - Could not connect to 10.25.13.30: Host 10.25.254.50 is not allowed to connect!
```

It would just make troubleshooting easier for the end user.

**Comment by kohenkatz** [comment]
@box293 I would think that this is a security benefit. If someone is scanning for open (perhaps vulnerable to some exploit) NRPE services, you'd rather give them a generic "connection closed" message than explain to them why the connection was closed. As it currently is, "Connection reset by peer" is your machine's way of saying "the remote server hung up on us." The actual text of that message has nothing to do with NRPE itself.

**Comment by box293** [comment]
I see the security side of things, but I also see how end users get frustrated troubleshooting problems like this, it's not very helpful.

It's tough to draw the line between security and user friendly.

**Comment by ericloyd** [comment]
I would agree that not giving out specific reasons for failure is a good security practice.  In theory, the person setting up NRPE on the remote host has access to the logs and can do what you just did, @box293.  Namely, look at the local logs to see the specific reason.  

This is the same reason a password failure message should be no more than "login failed" as opposed to "unknown user" or "invalid password."  The less you give out, the safer you are.
